### PR TITLE
iproute, netns: support netlink_groups

### PR DESCRIPTION
**Pull request by svinota**
Bug-Url: https://github.com/svinota/pyroute2/issues/995

### CODE REVIEW

File: pyroute2/iproute/linux.py [hunk 1]
```diff
@@ -60,6 +60,14 @@
     RTM_NEWTCLASS,
     RTM_NEWTFILTER,
     RTM_SETLINK,
+    RTMGRP_IPV4_IFADDR,
+    RTMGRP_IPV4_ROUTE,
+    RTMGRP_IPV4_RULE,
+    RTMGRP_IPV6_IFADDR,
+    RTMGRP_IPV6_ROUTE,
+    RTMGRP_IPV6_RULE,
```
Comment (ffourcot) [inline pyroute2/iproute/linux.py:68]: I do not found RTMGRP_IPV6_RULE constant in current kernel headers. Is it a typo with RTNLGRP_IPV6_RULE ? (RTM vs RTN)

Comment (ffourcot) [reply pyroute2/iproute/linux.py:68]: Ok, it looks like all RTMGRP_* variables are outdated and working only as compat.

See https://lwn.net/Articles/147608/ for reference

Comment (svinota) [reply pyroute2/iproute/linux.py:68]: There is already `NetlinkSocket.add_membership()/drop_membership()`, so probably I have to change NDB sources to use it. Let's schedule it as a separate task.

**Comment by cunha** [comment]
I went through the changes and they make sense to me. I also ran some local tests and can confirm that the new `netlink_groups` option significant improves NDB creation time. It covers my use-case successfully.

One thing to consider/document in the API is that we can pass in `netlink_groups` as a parameter when creating `IPRoute` and `NetNS` instances. The `netlink_groups` are later passed on to `IPRoute.bind()` and `NetNS.bind()`. However, `netlink_groups` are not passed on to `dump()`, which instead receives a `groups` parameter.

I can contribute changes to the docs to cover this parameter and its use around early September.

**Comment by cunha** [comment]
What I was getting at is that `dump()` does not inherit the `netlink_groups` passed to `__init__()`, while `bind()` does. I am not saying this is a problem, I just wanted to mention it for awareness.

I think we can leave everything as-is, including `bind(groups)` to avoid the deprecation. We just need to update the configuration to match the code (which I can do in a few weeks).

**Comment by svinota** [comment]
Yep, I tried now to change `groups` to `netlink_groups`, but it affects too much unrelated code.

`dump()` does not inherit `netlink_groups` 'cause simple dump commands (`link('dump')`, `route('dump')` etc) are not affected by the `bind()` arguments, they return the same results.

But you correctly point this out, it may be an issue, since it's another inconsistency.

**Comment by svinota** [comment]
Finally dropping `netlink_groups` in favour of `groups`. The reasons:

* constency with existing `bind()`
* the attribute is to be used by PF_ROUTE sockets as well, and PF_ROUTE ≠ netlink, thus `netlink_groups` is not the best naming

Also, made `dump()` to respect `self.groups` by default, if not forced by the method argument.

**Comment by svinota** [comment]
@cunha , @ffourcot — if no objections, I would merge.

**Comment by cunha** [comment]
Ran my (rather minimal) tests again without issues.

**Comment by ffourcot** [comment]
@svinota it looks great, thanks!

**Pull request merged**